### PR TITLE
Add question on unit testing

### PR DESCRIPTION
**Pull request by theIYD**
## Description

A new question on unit testing.

## What does your PR belong to?

* [x] Questions / Answers
* [ ] Website
* [ ] General / Things regarding the repository (like CI Integration)

## Types of changes

* [ ] Bug fix (non-breaking change which fixes an issue)
* [ ] Enhancement (non-breaking improvement of a question)
* [x] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

* [ ] My code follows the code style of this project.
* [ ] My change requires a change to the documentation.
* [ ] I have updated the documentation accordingly.
* [ ] I have checked that the changes are working properly
* [x] I have checked that there isn't any PR doing the same
* [x] I have read the **CONTRIBUTING** document.

### CODE REVIEW

File: questions/unit-testing.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+### What is Unit Testing ?
+
+### Answer
+
+In layman terms, unit Testing is a level of software testing where individual units or components of a software are tested independently.
```
Comment (skatcat31) [inline questions/unit-testing.md]:
```suggestion
Unit Testing is a level of software testing where individual units or components of software are tested independently.
```

File: questions/unit-testing.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+Generally, we write code tests using frameworks like Mocha and Chai. We validate the module by giving it all types of inputs into it, and expect the output we are looking for.
```
Comment (skatcat31) [inline questions/unit-testing.md]:
```suggestion
Tests are written using frameworks like Mocha, Chai, and Jest. The software is validated by giving it different inputs, and checking to see that the desired output is returned.
```

File: questions/unit-testing.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+**Benefits**:
```
Comment (skatcat31) [inline questions/unit-testing.md]: This section should be in Good to hear

File: questions/unit-testing.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+- Keeps the unit working.
+- The unit is always executed on any new change.
+
+### Good to hear
+
+Take a look at different testing frameworks like Jest which are way easier to use.
```
Comment (skatcat31) [inline questions/unit-testing.md]: Edited to mention Jest above.

File: questions/unit-testing.md [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
```
Comment (skatcat31) [inline questions/unit-testing.md]:
```suggestion
Test the software in a way the end user would be expected to use it.

```

File: questions/unit-testing.md (hunk 3 above)
Comment (theIYD) [reply questions/unit-testing.md]: The Benefits ?

Comment (skatcat31) [reply questions/unit-testing.md]: Yup. You wouldn't mention them as part of the answer, but it would be good to hear and even good to research on your own.

Comment (theIYD) [reply questions/unit-testing.md]: Done ! @skatcat31

File: questions/unit-testing.md [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+### Answer
+
+Unit Testing is a level of software testing where individual units or components of software are tested independently.
+
+Tests are written using frameworks like Mocha, Chai, and Jest. The software is validated by giving it different inputs, and checking to see that the desired output is returned.
+
+### Good to hear
+
+**Benefits**:
```
Comment (skatcat31) [inline questions/unit-testing.md]: you can remove the benefits bold here entirely now, since the benefits are the good to hear

File: questions/unit-testing.md [hunk 7]
```diff
@@ -0,0 +1,22 @@
+### What is Unit Testing ?
+
+### Answer
+
+Unit Testing is a level of software testing where individual units or components of software are tested independently.
+
+Tests are written using frameworks like Mocha, Chai, and Jest. The software is validated by giving it different inputs, and checking to see that the desired output is returned.
+
+### Good to hear
```
Comment (skatcat31) [inline questions/unit-testing.md]: The good to hear section needs some reformatting to look like other submissions

**Comment by fejes713** [comment]
Closing due to inactivity.
